deepspeed/runtime/state_dict_factory.py

```diff
@@ -25,19 +28,24 @@ def get_sd_loader_json(json_file):
             return SDLoaderFactory.get_sd_loader(ckpt_list, sd_type, version)
 
     @staticmethod
-    def get_sd_loader(ckpt_list, sd_type='Megatron', version=None):
+    def get_sd_loader(ckpt_list,
+                      sd_type='Megatron',
+                      version=None,
+                      checkpoint_engine=None):
         if sd_type == 'Megatron':
-            return MegatronSDLoader(ckpt_list, version)
+            return MegatronSDLoader(ckpt_list, version, checkpoint_engine)
         else:
             assert False, '{} checkpoint type is not supported'.format(sd_type)
 
 
 class SDLoaderBase(ABC):
-    def __init__(self, ckpt_list, version):
+    def __init__(self, ckpt_list, version, checkpoint_engine=None):
```

Comment:
Make `checkpoint_engine` required. This will help to avoid unexpected creation of CheckpointEngine objects at runtime.